packages/adapter/src/lib/adapter/utils.ts

```diff
@@ -260,4 +260,63 @@ export class Utils {
         }
         return result;
     }
+
+    /**
+     * Validates the object-type argument that is passed to setState
+     * @param obj object to validate
+     */
+    validateSetStateObjectArgument(obj: Record<string, any>): void {
+        // Check that we have at least one existing non-undefined property at all, else invalid
+        if (!Object.keys(obj).some(key => obj[key] !== undefined)) {
```

Comment:
```suggestion
        if (!Object.values(obj).some(prop => prop !== undefined)) {
```

Reply:
> Holy shit, it's happening 🙈

But I think tests are failing now. Have to look into it where I messed it up with receiving messages